backend/src/main/java/com/google/step/servlets/WebhookServlet.java

```diff
@@ -67,9 +60,21 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       return; //stop execution, we expect an id param in the url
     }
     Key advertiserKey = KeyFactory.stringToKey(advertiserKeyString);
-
+    User user;
+    try {
+      user = AdvertiserUtil.getUserFromAdvertiserKey(advertiserKey);
+    } catch (EntityNotFoundException e) {
+      response.sendError(404,
+          "Invalid webhook url. Advertiser has not registered with our site.");
+      return;
+    }
     myLead = Lead.fromReader(request.getReader());
     //TODO: Add additional verification steps
+    try {
+      EmailUtil.sendNewLeadEmail(user);
+    } catch (MessagingException e) {
+      System.out.println(e);
```

Comment:
What behavior do we want if this fails? Do we want to notify that email failed, or maybe retry (with a limit)?

Reply:
Hm, I'm not sure what we want to do.
I guess for now, we can log that the email failed and decide whether or not we want to retry?